test/e2e/kubeproxy.go

```diff
@@ -0,0 +1,447 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"io/ioutil"
+	api "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/latest"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	endpointHttpPort        = 8080
+	endpointUdpPort         = 8081
+	endpointHostPort        = 8082
+	testContainerHttpPort   = 8080
+	clusterHttpPort         = 80
+	clusterUdpPort          = 90
+	nodeHttpPort            = 32080
+	nodeUdpPort             = 32081
+	loadBalancerHttpPort    = 100
+	imageName               = "artfulcoder/netproxy:1.0"
+	testPodName             = "test-container-pod"
+	nodePortServiceName     = "node-port-service"
+	loadBalancerServiceName = "load-balancer-service"
+)
+
+type TestConfig struct {
+	testContainerPod    *api.Pod
+	testHostPod         *api.Pod
+	endpointPods        []*api.Pod
+	f                   *Framework
+	nodePortService     *api.Service
+	loadBalancerService *api.Service
+	nodes               []string
+}
+
+var _ = Describe("KubeProxy", func() {
+	f := NewFramework("e2e-kubeproxy")
+	config := &TestConfig{
+		f: f,
+	}
+	It("should test kube-proxy", func() {
+
+		By("cleaning up any pre-existing namespaces used by this test")
+		f.cleanup()
+
+		By("creating a selector")
```

Comment:
No, By() is just progress marks. http://onsi.github.io/ginkgo/#documenting-complex-its-by